pkg/verify/log_root.go

```diff
@@ -0,0 +1,64 @@
+//
+// Copyright 2021 The Sigstore Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package verify
+
+import (
+	"crypto"
+	"crypto/ecdsa"
+	"fmt"
+
+	"github.com/google/trillian/types"
+	"github.com/pkg/errors"
+)
+
+// this verification copied from https://github.com/google/trillian/blob/v1.3.13/crypto/verifier.go
+// which has since been deleted
+
+// SignedLogRoot verifies the signed log root and returns its contents
+func SignedLogRoot(pub crypto.PublicKey, hash crypto.Hash, logRoot, logRootSignature []byte) (*types.LogRootV1, error) {
```

Comment:
nit: if we're hardcoding the hash to SHA256, is there a reason why we're exposing this as an argument here?

Reply:
nope, removed